lightningd/peer_control.c

```diff
@@ -1774,7 +1774,7 @@ void update_channel_from_inflight(struct lightningd *ld,
 	channel->funding_sats = inflight->funding->total_funds;
 
 	channel->our_funds = inflight->funding->our_funds;
-	channel->our_funds.satoshis += inflight->funding->splice_amnt;
+	channel->our_funds.satoshis += inflight->funding->splice_amnt; /* Raw: splicing */
```

Comment:
amount_sat_add(&channel->our_funds, channel->our_funds, amount_sat(inflight->funding->splice_amnt)) ?

Then you figure out how to panic if that overflows!

Reply:
amount.h
```
WARN_UNUSED_RESULT bool amount_sat_add_sat_s64(struct amount_msat *val,
					       struct amount_msat a,
					       struct s64 b);
```

amount.c
```
WARN_UNUSED_RESULT bool amount_sat_add_sat_s64(struct amount_msat *val,
					       struct amount_msat a,
					       struct s64 b)
{
	if (b < 0)
		return amount_sat_sub_sat(val, a, amount_sat(-b));
	else
		return amount_sat_add_sat(val, a, amount_sat(b));
}
```

peer_control.c
```
	if (!amount_sat_add_sat_s64(&channel->our_funds, channel->our_funds,
				    inflight->funding->splice_amnt)) {

		channel_fail_permanent(channel,
				       REASON_LOCAL,
				       "Updaing channel view for splice causes"
				       " an invalid satoshi amount wrapping,"
				       " channel: %s, initial funds: %s, splice"
				       " banace change: %s",
				       type_to_string(tmpctx, struct channel_id,
						      &channel->cid),
				       type_to_string(tmpctx, struct amount_sat,
						      &our_funds),
				       inflight->funding->splice_amnt);
	}
```